### PR TITLE
[SPARK-45568][TESTS] Fix flaky WholeStageCodegenSparkSubmitSuite

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

WholeStageCodegenSparkSubmitSuite is [flaky](https://github.com/apache/spark/actions/runs/6479534195/job/17593342589) because SHUFFLE_PARTITIONS(200) creates 200 reducers for one total core and improper stop progress causes executor launcher reties. The heavy load and reties might result in timeout test failures.

### Why are the changes needed?

CI robustness

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

existing WholeStageCodegenSparkSubmitSuite 
### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/execution/WholeStageCodegenSparkSubmitSuite.scala [hunk 1]
```diff
@@ -70,39 +71,41 @@ class WholeStageCodegenSparkSubmitSuite extends SparkSubmitTestUtils
...
-    spark = SparkSession.builder().getOrCreate()
+    val spark = SparkSession.builder()
+      .config(SQLConf.SHUFFLE_PARTITIONS.key, "2")
+      .getOrCreate()
+
+    try{
+      // Make sure the test is run where the driver and the executors uses different object layouts
+      val driverArrayHeaderSize = Platform.BYTE_ARRAY_OFFSET
+      val executorArrayHeaderSize =
+        spark.sparkContext.range(0, 1).map(_ => Platform.BYTE_ARRAY_OFFSET).collect.head.toInt
```
Comment (LuciferYang) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/WholeStageCodegenSparkSubmitSuite.scala:85]: we can remove this `toInt`, it is redundant

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Comment by LuciferYang** [comment]
Is this also an flaky test for other branches? Does it need to be backported?

**Comment by yaooqinn** [comment]
> Is this also a flaky test for other branches? Does it need to be backported?


yes

**Comment by yaooqinn** [comment]
thanks all, merged to master 3.5 3.4 and 3.3

**Comment by dongjoon-hyun** [comment]
Thank you, @yaooqinn and all!

**Pull request closed without merging**